File: [23.12]Design-Pattern/Java/src/structural/proxy/ProtectiveProxy.java
package structural.proxy;

public class ProtectiveProxy implements Image{
    String path;
    String authority;

    public ProtectiveProxy(String path, String authority) {
        this.path = path;
        this.authority = authority;
        System.out.println("["+path +" 경로, "+authority+"사용자] 프록시 생성");
    }

    @Override
    public void showImage() {
        if(this.authority.equals("관리자")){
            System.out.println("관리자 접근");
            HighImage highImage = new HighImage(this.path);
            highImage.showImage();
        }else {
            System.out.println(this.authority + "는 접근할 수 없습니다.");
        }
    }



}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/structural/proxy/ProtectiveProxy.kt
package structural.proxy

class ProtectiveProxy(val path: String, val user: String): Image{
    init{
        println("$path 경로 프록시 생성")
    }

    override fun showImage() {
        when(this.user){
            "관리자" -> {
                println("$user 접근 성공")
                val highImage = HighImage(this.path)
                highImage.showImage()
            }
            else -> println("$user 접근 불가")
        }


    }

}